src/Combinatorics/Graphs/functions.jl

```diff
@@ -2,9 +2,50 @@ import Oscar: Polyhedron, Polymake, pm_object
 import Oscar.Polymake: Directed, Undirected
 
 function pm_object(G::Graph{T}) where {T <: Union{Directed, Undirected}}
-    return G.pm_graph
+  return G.pm_graph
 end
 
+@doc raw"""
+    directed_component(G::MixedGraph)
+
+Return the directed graph that is the subgraph of the graph `G`.
```

Comment:
I think this docstring should contain a warning that the returned graph may not be modified, as that messes up your assumptions about the mixed graph.
Example: I have a mixed graph with 2 vertices (aka both components have 2 verts). A user may now call `add_vertex!(directed_component(G))`, thus breaking an internal contract of the MixedGraph type.
So you either need to mention this behaviour here in the docstring, or let this function make a copy and have an internal non-copying helper function for use in all your other functions.

Reply:
good point